nexus/db-queries/src/db/datastore/external_ip.rs

```diff
@@ -239,29 +222,7 @@ impl DataStore {
     ) -> CreateResult<ExternalIp> {
         let ip_id = Uuid::new_v4();
 
-        // This implements the same pattern as in `allocate_instance_ephemeral_ip` to
-        // check that a chosen pool is valid from within the current silo.
-        let pool = match pool {
-            Some(authz_pool) => {
-                let (.., pool) = LookupPath::new(opctx, &self)
-                    .ip_pool_id(authz_pool.id())
-                    .fetch_for(authz::Action::CreateChild)
-                    .await?;
```

Comment:
this fetch was not necessary at all — replaced by `opctx.authorize` call in `resolve_pool_for_allocation`